solana/allocator/src/lib.rs

```diff
@@ -21,9 +25,9 @@ use core::cell::Cell;
 /// a segfault once out of available heap memory.
 pub struct BumpAllocator {
     #[cfg(test)]
-    start: core::ptr::NonNull<Cell<usize>>,
+    ptr: core::ptr::NonNull<u8>,
```

Comment:
why are these attributes only allowed only in tests?

Reply:
In production we’re using Solana provided heap.  `ptr` is only for tests where we allocate chunk of memory and use that.